calabash-cucumber/lib/calabash-cucumber/launch/simulator_launcher.rb

```diff
@@ -251,13 +251,27 @@ def app_bundle_or_raise(path=nil, device_build_dir='iPhoneSimulator')
             puts('-'*37)
           end
         else
-          bo_dir = build_output_dir_for_project
-          sim_dirs = Dir.glob(File.join(bo_dir, '*-iphonesimulator', '*.app'))
+          search_dir = build_output_dir_for_project || DERIVED_DATA
+          sim_dirs = ''
+          apps = `find #{search_dir} -type d -name "*.app" -exec stat -f "%m %N" {} \\; | sort -rn | cut -d" " -f2`.split("\n")
+          apps.each do |app_path|
+            lipo = RunLoop::Lipo.new(app_path)
+            arches = lipo.info
+            if arches.include?("x86_64") || arches.include?("i386")
+              app =  RunLoop::App.new(app_path)
+              executable_name = app.executable_name
+              path_to_bin = File.join(app_path, executable_name)
+              if `xcrun strings "#{path_to_bin}" | grep -E 'CALABASH VERSION'`.include? "CALABASH VERSION"
```

Comment:
@jmody thanks for feedback, since that changes is in run_loop should i worried about it versions? I mean if we will have old run_loop installed without  
`app.calabash_server_version` method new implementation will fail in that case.
And i think i should update it to `find` instead of `each`.